src/main/java/org/zalando/nakadi/config/NakadiConfig.java

```diff
@@ -67,4 +73,24 @@ public MetricRegistry metricRegistry() {
         return metricRegistry;
     }
 
+    @Bean
+    public SystemProperties systemProperties(final ApplicationContext context) {
+        return name -> context.getEnvironment().getProperty(name);
+    }
+
+    @Bean
+    @SuppressWarnings("unchecked")
+    public ApplicationService factory(@Value("${nakadi.auth.plugin.factory}") final String factoryName,
```

Comment:
we create ApplicationService from ApplicationServiceFactory here, maybe we can change it to smth like
` public ApplicationService applicationService`